### PR TITLE
Change data structure of filters saved and use in filter form

### PR DESCRIPTION
**Pull request by AdityaKhatri**
Depends on: https://github.com/the-deep/server/pull/722

### CODE REVIEW

File: src/views/AnalysisModule/Analysis/index.tsx [hunk 1]
```diff
@@ -120,8 +128,12 @@ function Analysis(props: ComponentProps) {
         },
     );
 
+    useEffect(() => {
```
Comment (tnagorra) [inline src/views/AnalysisModule/Analysis/index.tsx:133]: Please add a note why we are fetching/re-triggering on modifedAt change

File: src/views/AnalysisModule/Analysis/index.tsx [hunk 2]
```diff
@@ -131,6 +143,7 @@ function Analysis(props: ComponentProps) {
             method: 'DELETE',
             onSuccess: () => {
                 pillarGetTrigger();
+                onAnalysisPillarDelete();
```
Comment (tnagorra) [inline src/views/AnalysisModule/Analysis/index.tsx:147]: Shouldn't this be before pillarGetTrigger

File: src/views/AnalysisModule/AnalysisEditModal/index.tsx [hunk 3]
```diff
@@ -190,9 +193,26 @@ function AnalysisEditModal(props: AnalysisEditModalProps) {
...
+        const matrixMap = listToMap(
+            matrixPillars,
+            idSelector,
+            d => ({
+                id: d.id,
+                key: d.key,
+                uniqueId: d.uniqueId,
+            }),
+        );
+        const newAnalysisPillar = [...analysisPillar].map(ap => ({
```
Comment (tnagorra) [inline src/views/AnalysisModule/AnalysisEditModal/index.tsx]:
```suggestion
        const newAnalysisPillar = analysisPillar.map(ap => ({
```

File: src/views/AnalysisModule/AnalysisEditModal/index.tsx [hunk 4]
```diff
@@ -190,9 +193,26 @@ function AnalysisEditModal(props: AnalysisEditModalProps) {
 
     const onValidationSuccess = useCallback((finalValues) => {
```
Comment (tnagorra) [inline src/views/AnalysisModule/AnalysisEditModal/index.tsx:196]: Please add typings for finalValues

File: src/views/PillarAnalysis/index.tsx [hunk 5]
```diff
@@ -100,6 +104,17 @@ function PillarAnalysis(props: PageProps) {
...
+        if (isDefined(pillarAnalysis)) {
+            const newFilters = listToGroupList(
+                pillarAnalysis.filters,
+                o => o.key,
+                o => o.id,
+            );
+            setFiltersValue(newFilters);
+        }
+    }, [pillarAnalysis]);
+
```
Comment (tnagorra) [inline src/views/PillarAnalysis/index.tsx]: Why not add this logic on onCompleted?

**Review by tnagorra — CHANGES_REQUESTED** [review]
Needs some re-work!

File: src/views/AnalysisModule/Analysis/AnalysisPillar/index.tsx [hunk 6]
```diff
@@ -40,9 +42,10 @@ function AnalysisPillar(props: ComponentProps) {
         projectId,
         analysisId,
         createdAt,
+        pendingPillarDelete,
```
Comment (samshara) [inline src/views/AnalysisModule/Analysis/AnalysisPillar/index.tsx:45]: this variable is unused

File: src/views/AnalysisModule/Analysis/AnalysisPillar/index.tsx [hunk 7]
```diff
@@ -40,9 +42,10 @@ function AnalysisPillar(props: ComponentProps) {
...
     } = props;
 
-    const [completed, setCompleted] = useState(false);
+    const [completed] = useState(false);
     // setIsCompleted to be used when the status is passed by API
```
Comment (samshara) [inline src/views/AnalysisModule/Analysis/AnalysisPillar/index.tsx]: can't we just use declare completed as false here

**Pull request merged**